spring-boot/src/main/java/org/springframework/boot/jta/narayana/NarayanaRecoveryManagerBean.java

```diff
@@ -41,8 +40,8 @@ public NarayanaRecoveryManagerBean(RecoveryManagerService recoveryManagerService
 		this.recoveryManagerService = recoveryManagerService;
 	}
 
-	@Override
-	public void afterPropertiesSet() throws Exception {
+	@EventListener
+	public void create(ApplicationReadyEvent ignored) throws Exception {
```

Comment:
Why would there be any transaction logic happening before the application has started?